lib/bundix/shell_nix_context.rb

```diff
@@ -0,0 +1,28 @@
+class Bundix
+  class ShellNixContext < Struct.new(:project, :ruby, :gemfile, :lockfile, :gemset)
+    def self.from_hash(hash)
+      p, r, gf, l, gs = hash.values_at(:project, :ruby, :gemfile, :lockfile, :gemset)
+      self.new(p,r,gf,l,gs)
+    end
+
+    def bind
+      binding
+    end
+
+    def path_for(file)
+      "./#{Pathname(file).relative_path_from(Pathname('./'))}"
```

Comment:
why is the `./` prefix necessary?

Reply:
They're templated as Nix path values, so they have to have a "/". Easy enough to add a guard for "no /" and prefix only if needed.

Reply:
If I were to change this I would return a Pathname here, change the ruby2nix generator to convert Pathname to nix paths and finally use it in the template.

I'll leave it up to you if you want to do that.

Reply:
Oh, that's a really good idea. Let me chip that into this PR.

Reply:
ping me when you're done with the revision